velox/type/StringView.h

```diff
@@ -77,6 +76,11 @@ struct StringView {
     }
   }
 
+  static StringView makeInline(std::string str) {
+    VELOX_CHECK(isInline(str.size()));
```

Comment:
@bkietz @mbasmanova @Yuhta should we turn this into a VELOX_DCHECK instead? I like that we're safer with this check and clients explicitly marking it as inline, but we've seen this macro preventing functions from being inlined. Thoughts?

Reply:
I think we should use `DCHECK` here